pingdom/resource_pingdom_maintenance_window.go

```diff
@@ -0,0 +1,242 @@
+package pingdom
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/russellcardullo/go-pingdom/pingdom"
+)
+
+func resourcePingdomMaintenanceWindow() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePingdomMaintenanceWindowCreate,
+		Read:   resourcePingdomMaintenanceWindowRead,
+		Update: resourcePingdomMaintenanceWindowUpdate,
+		Delete: resourcePingdomMaintenanceWindowDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"from": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: false,
+			},
+			"to": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: false,
+			},
+			"recurrence_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "none",
+			},
+			"repeat_every": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"effective_to": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"uptimeids": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: false,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+			},
+			"tmsids": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: false,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+			},
+		},
+	}
+}
+
+type commonMaintenanceWindowParams struct {
+	Description   string
+	From          int64
+	To            int64
+	RecurenceType string
+	RepeatEvery   int
+	EffectiveTo   int
+	UptimeIds     []int
+	TmsIds        []int
+}
+
+func maintenanceWindowForResource(d *schema.ResourceData) (*pingdom.MaintenanceWindow, error) {
+	windowParams := commonMaintenanceWindowParams{}
+
+	if v, ok := d.GetOk("description"); ok {
+		windowParams.Description = v.(string)
+	}
+	if v, ok := d.GetOk("from"); ok {
+		windowParams.From = v.(int64)
+	}
+	if v, ok := d.GetOk("to"); ok {
+		windowParams.To = v.(int64)
```

Comment:
I get a panic on this line: `panic: interface conversion: interface {} is int, not int64`.  I think this and the `from` param should probably be changed to `int` as the terraform SDK only has `TypeInt`.